src/value.c

```diff
@@ -100,6 +124,36 @@ inline int SIValue_IsNullPtr(SIValue *v) {
 	return v == NULL || v->type == T_NULL;
 }
 
+int SIArray_ToString(SIValue list, char *buf, size_t len) {
+	if(len <= 1) return 0;
+	int bytes_written = snprintf(buf, len, "[");
+	len -= bytes_written;
+	uint arrayLen = array_len(list.array);
+	for(uint i = 0; i < arrayLen; i ++) {
+		if(len < 2) break;
+
+		// write the next value
+		int currentWriteLength = SIValue_ToString(list.array[i], buf + bytes_written, len);
+		bytes_written += currentWriteLength;
+		len -= currentWriteLength;
+
+		// if there no more space in the buffer of the end of the array
+		if(len < 2 || i == arrayLen - 1) break;
+
+		currentWriteLength = snprintf(buf + bytes_written, len, ", ");
+		bytes_written += currentWriteLength;
+		len -= currentWriteLength;
+	}
+	// you can close the array
+	if(len >= 2) {
```

Comment:
we can't require this since even though the array length is known, the length of the content is known only after we evaluate each value to a string.
[10,10,10] is the same size and type of [10000000,10000000,10000000], for e given buffer with a length of 12 only the first array will fit, and the second will be known not to be fit only after the second value "toString" call.

Reply:
done
```
// minimum length buffer for "[...]"
if(len <= 5) return 0;
```